airbyte-cdk/python/unit_tests/sources/declarative/requesters/paginators/test_page_increment.py

```diff
@@ -35,3 +36,16 @@ def test_page_increment_paginator_strategy(page_size, start_from, last_records,
 
     paginator_strategy.reset()
     assert start_from == paginator_strategy._page
+
+
+@pytest.mark.parametrize(
+    "inject_on_first_request, expected_initial_token",
+    [
+        pytest.param(True, 0, id="test_with_inject_offset"),
```

Comment:
nit: can you also add a test where we do inject the offset, but the initial offset is not 0 to assert this works with `start_from_page`